java/ql/lib/semmle/code/java/regex/regex.qll

```diff
@@ -0,0 +1,943 @@
+import java
+private import RegexFlowConfigs
+
+/**
+ * A string literal that is used as a regular exprssion.
+ */
+abstract class RegexString extends Expr {
+  RegexString() { this instanceof StringLiteral }
+
+  /** Holds if a character set starts between `start` and `end`. */
+  private predicate char_set_start0(int start, int end) {
```

Comment:
snake -> camel case everywhere

Reply:
The [QL style guide](https://github.com/github/codeql/blob/main/docs/ql-style-guide.md) specifies camel case for predicate names.